sycl/test-e2e/Basic/accessor/accessor.cpp

```diff
@@ -1074,6 +1074,29 @@ int main() {
     assert(Data == 64);
   }
 
+  // iterator operations test for 0-dim buffer accessor
+  {
+    sycl::queue Queue;
+    int Data[] = {32, 32};
+
+    // Explicit block to prompt copy-back to Data
+    {
+      sycl::buffer<int, 1> DataBuffer(Data, sycl::range<1>(2));
+
+      Queue.submit([&](sycl::handler &CGH) {
+        sycl::accessor<int, 0> Acc(DataBuffer, CGH);
+        CGH.single_task<class acc_0_dim_iter_assignment>([=]() {
+          *Acc.begin() = 64;
+          *Acc.rbegin() += 64;
```

Comment:
I'm confused a little. A 0-dim accessor should have an access to only one element, isn't it?

Reply:
@KornevNikita 
In this case, if we had 1 dimensional accessor, `*Acc.rbegin()` would access the second element of the buffer. Because it's a 0-dim, both `rbegin` and `begin` point to the same element.

Reply:
Oh, really. Thanks for clarification.